stories/index.js

```diff
@@ -1,5 +1,65 @@
 // @flow
+import { storiesOf } from '@storybook/react'
+import { withKnobs } from '@storybook/addon-knobs'
+import { utils } from '../src'
 import '../src/index.css'
-import './grid'
-import './components'
-import './layout'
+
+/**
+ * Reads the `/components`, `/grid` and `/layout` subfolder files and adds them to their respective
+ * stories
+ */
+
+const folders = {
+  './components': require.context('./components', true, /.+\.js$/i),
+  './grid': require.context('./grid', true, /.+\.js$/i),
+  './layout': require.context('./layout', true, /.+\.js$/i),
+}
+
+function unwrapDisplayNameFromHOC(name: string = '') {
+  return (name.match(/(?:\(([^()]+)\))/) || [])[1] || name
+}
+
+function configStories(storiesOfName: string, storiesModule: typeof module) {
+  const stories = storiesOf(storiesOfName, storiesModule)
+
+  stories.addDecorator(withKnobs)
+
+  const addStoryWithJSX = (storyName: string, ComponentFn: () => any) =>
+    stories.addWithJSX(storyName, ComponentFn, {
+      showDefaultProps: false,
+      useBooleanShorthandSyntax: true,
+      skip: 1, // assumes each story will have a WithExtensions wrapper, this config option skips that wrapper
+      displayName: element =>
+        unwrapDisplayNameFromHOC(utils.getDisplayName(element.type)),
```

Comment:
the function I had here was slightly different from getDisplay name, which is why I didn't use it originally. This is because we also need to account for primitives (h1, div, p, etc) with `element.type.name`. Although maybe we need to update the util function anyways to account for that.